### PR TITLE
Rename tokenset and mrtap config functions to be async

### PR DESCRIPTION
**Pull request by lgmugnier**
## Description of the change

We need to rename the functions that handle the tokenset and mrtap configs to be async because we need them to be for windows integration. It's a simple change but it requires some work and so I put it in a separate PR for ease of review

## Testing

Describe how to test this PR....

**backend branch:** 
**bzero branch:** 
**charts branch:** 

#### Ready to run system tests?

- [ ] Yes

## Relevant release note information

Release Notes:

## Related JIRA tickets

Relates to JIRA: CWC-2516

## Relevant HotFix if applicable

Relevant HotFix PR Number:

## Have you considered the security impacts?

Does this PR have any security impact?

- [ ] Yes
- [ ] No

If yes, please explain:

Does this PR add a new/modifies the name/arguments of a `zli` command?

- [ ] Yes
- [ ] No

If yes, please provide the docs/zli-reference-manual branch reflecting the changes:

### CODE REVIEW

File: src/services/oauth/oauth.service.ts [hunk 1]
```diff
@@ -329,11 +329,11 @@ export class OAuthService implements IDisposable {
      * @returns The current user's id_token
      */
     public async getIdToken(): Promise<string> {
-        const tokenSet = this.configService.getTokenSet();
+        const tokenSet = await this.configService.getTokenSet();
 
         // Refresh if the token exists and has expired
         if (tokenSet) {
-            if (!this.isAuthenticated()) {
+            if (!await this.isAuthenticated()) {
```
Comment (johncmerfeld) [inline src/services/oauth/oauth.service.ts:336]: Wait, does this work? I didn't realize you could use `await` inside an `if` clause

File: src/services/http/http.service.ts [hunk 2]
```diff
@@ -230,7 +230,7 @@ export class HttpService {
     }
 
     protected async FormPost<TReq, TResp>(route: string, body: TReq): Promise<TResp> {
-        this.setHeaders();
+        await this.setHeaders();
```
Comment (johncmerfeld) [inline src/services/http/http.service.ts:233]: Did you intend to include this change in this PR? It's neither `mrtap` nor `tokenset`

File: src/services/config/config.service.ts [hunk 3]
```diff
@@ -293,12 +269,14 @@ export class ConfigService implements IKubeDaemonSecurityConfigService, IKubeCon
...
+    async getIdToken(): Promise<string> {
+        const tokenSet = await this.config.getTokenSet();
 
-    getAccessToken(): string {
-        return this.config.getTokenSet().access_token;
+        try {
+            return tokenSet.id_token;
+        } catch {
+            return '';
+        }
```
Comment (johncmerfeld) [inline src/services/config/config.service.ts:279]:
```suggestion
        return tokenSet ? tokenSet.id_token : '';
```

File: src/handlers/login/login.handler.ts [hunk 4]
```diff
@@ -146,14 +146,14 @@ export async function loginServiceAccountHandler(configService: ConfigService, l
     const oAuthService = new OAuthService(configService, logger);
 
     let bzeroCredsFile: ServiceAccountBzeroCredentials = null;
-    if (!oAuthService.isAuthenticated()) {
+    if (!await oAuthService.isAuthenticated()) {
```
Comment (johncmerfeld) [inline src/handlers/login/login.handler.ts:149]: Same question as below. Just surprised this compiles

File: src/services/oauth/oauth.service.ts (hunk 1 above)
Comment (lgmugnier) [reply src/services/oauth/oauth.service.ts:336]: as far as I've tested, this does work

Comment (tflpd) [reply src/services/oauth/oauth.service.ts:336]: confirming this works in JS

File: src/handlers/login/login.handler.ts [hunk 5]
```diff
@@ -71,13 +71,13 @@ export async function loginUserHandler(configService: ConfigService, logger: Log
     // Can only create oauth service after loginSetup completes
     const oAuthService = new OAuthService(configService, logger);
 
-    if (!oAuthService.isAuthenticated()) {
+    if (!await oAuthService.isAuthenticated()) {
         // Create our Nonce
         const nonce = mrtapService.createNonce();
         // Pass it in as we login
-        await oAuthService.login((t) => {
+        await oAuthService.login(async (t) => {
```
[the remark this answers is not in the document]
Comment (lgmugnier) [reply src/handlers/login/login.handler.ts:78]: 64a852c

File: src/services/config/config.service.ts [hunk 6]
```diff
@@ -293,12 +269,9 @@ export class ConfigService implements IKubeDaemonSecurityConfigService, IKubeCon
...
 
-    getIdToken(): string {
-        return this.config.getTokenSet().id_token;
-    }
-
-    getAccessToken(): string {
-        return this.config.getTokenSet().access_token;
+    async getIdToken(): Promise<string> {
+        const tokenSet = await this.config.getTokenSet();
+        return tokenSet ? tokenSet.id_token : '';
```
[the remark this answers is not in the document]
Comment (lgmugnier) [reply src/services/config/config.service.ts]: 64a852c

**Review by tflpd — APPROVED** [review]

**Pull request merged**